apps/total-typescript/src/trpc/routers/bookmarks.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  addBookmark,
+  BookmarkInputSchema,
+  ReturnedBookmarkSchema,
+  deleteBookmark,
+  getBookmarksForUser,
+  getBookmarkByResourceId,
+  getLastBookmarkedResource,
+} from '@/lib/bookmarks'
+import {publicProcedure, router} from '@skillrecordings/skill-lesson'
+import {getToken} from 'next-auth/jwt'
+import {z} from 'zod'
+
+export const bookmarksRouter = router({
+  getBookmark: publicProcedure
+    .input(z.object({id: z.string()}))
+    .query(async ({ctx, input}) => {
+      const token = await getToken({req: ctx.req})
+      if (!token) {
+        throw new Error('Unauthorized')
+      }
+
+      const bookmark = await getBookmarkByResourceId({
+        userId: token.id as string,
+        resourceId: input.id,
+      })
+
+      return ReturnedBookmarkSchema.parse(bookmark)
```

Comment:
parse the returned bookmark with zod for friendlier types. Prisma returns a Prisma.JsonValue for section and resource which isn't great to work with